src/components/ValueAnnotationTabs.tsx

```diff
@@ -0,0 +1,119 @@
+import { Paper, Tab, Tabs } from '@mui/material';
+import { useState, useEffect } from 'react';
+import { Columns } from '../utils/types';
+import Categorical from './Categorical';
+import Continuous from './Continuous';
+
+interface ValueAnnotationTabsProps {
+  columns: Columns;
+  dataTable: Record<string, string[]>;
+  onUpdateDescription: (columnID: string, value: string, description: string) => void;
+  onUpdateUnits: (columnID: string, units: string) => void;
+  onToggleMissingValue: (columnID: string, value: string, isMissing: boolean) => void;
+  onUpdateFormat: (columnID: string, format: { termURL: string; label: string } | null) => void;
+  onUpdateLevelTerm: (
+    columnID: string,
+    value: string,
+    term: { identifier: string; label: string } | null
+  ) => void;
+}
+
+function ValueAnnotationTabs({
+  columns,
+  dataTable,
+  onUpdateDescription,
+  onUpdateUnits,
+  onToggleMissingValue,
+  onUpdateFormat,
+  onUpdateLevelTerm,
+}: ValueAnnotationTabsProps) {
+  const [selectedColumnId, setSelectedColumnId] = useState<string | null>(null);
+  const columnEntries = Object.entries(columns);
+  const columnIds = Object.keys(columns);
+
+  useEffect(() => {
+    if (columnIds.length > 0 && !selectedColumnId) {
+      setSelectedColumnId(columnIds[0]);
+    }
+  }, [columnIds, selectedColumnId]);
+
+  useEffect(() => {
+    if (selectedColumnId && !columnIds.includes(selectedColumnId)) {
+      setSelectedColumnId(columnIds[0] || null);
+    }
+  }, [columnIds, selectedColumnId]);
```

Comment:
I think this could be a bit easier to understand and read if you handled it in a single useEffect and in there, handled both `setSelectedColumnId(columnIds[0])` together. 

i.e.
if you have column IDs -> 
  - if no column selected OR selected column is not in column IDs -> set selected column to first column ID
  - else -> set column to nothing

because the case of "have a column selected by have no column IDs" is included in the above.